model/switch_state.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"context"
+	"reflect"
+
+	val "github.com/serverlessworkflow/sdk-go/v2/validator"
+
+	"github.com/go-playground/validator/v10"
+)
+
+func init() {
+	val.GetValidator().RegisterStructValidationCtx(SwitchStateStructLevelValidation, SwitchState{})
+	val.GetValidator().RegisterStructValidationCtx(DefaultConditionStructLevelValidation, DefaultCondition{})
+	val.GetValidator().RegisterStructValidationCtx(EventConditionStructLevelValidation, EventCondition{})
+	val.GetValidator().RegisterStructValidationCtx(DataConditionStructLevelValidation, DataCondition{})
+}
+
+// SwitchState is workflow's gateways: direct transitions onf a workflow based on certain conditions.
+type SwitchState struct {
+	// TODO: we didn't use BaseState for this, because there is a lot of field which
+	// SwitchState didn't have.
+	BaseState
+
+	// Default transition of the workflow if there is no matching data conditions. Can include a transition or end definition
+	// Required
```

Comment:
it shouldn't be omitted if empty if the field is required.

Reply:
done, PTAL